datadog/resource_datadog_logs_pipeline.go

```diff
@@ -0,0 +1,687 @@
+			sourceRemapper.Sources = make([]string, len(sources))
+			for i, source := range sources {
+				sourceRemapper.Sources[i] = source.(string)
+			}
+		}
+		ddProcessor.Definition = sourceRemapper
+	case datadog.GrokParserType:
+		var grokParser = datadog.GrokParser{}
+		if source, ok := tfProcessor["source"].(string); ok {
+			grokParser.Source = &source
+		}
+		if grok, ok := tfProcessor["grok"].([]interface{}); ok && len(grok) > 0 {
+			var ddGrok = datadog.GrokRule{}
+			var tfG = grok[0].(map[string]interface{})
+			if support, ok := tfG["support_rules"].(string); ok {
+				ddGrok.SupportRules = &support
+			}
+			if match, ok := tfG["match_rules"].(string); ok {
+				ddGrok.MatchRules = &match
+			}
+			grokParser.GrokRule = &ddGrok
+		}
+		ddProcessor.Definition = grokParser
+	case datadog.NestedPipelineType:
+		var nestedPipeline = datadog.NestedPipeline{}
+		if filter, ok := tfProcessor["filter"].([]interface{}); ok && len(filter) > 0 {
+			ddFilter := datadog.FilterConfiguration{}
+			convertFilter(filter[0], &ddFilter)
+			nestedPipeline.Filter = &ddFilter
+		}
+		if processors, ok := tfProcessor["processor"].([]interface{}); ok {
+			nestedProcessors := make([]datadog.LogsProcessor, len(processors))
+			if err := convertTFProcessors(nestedProcessors, processors); err != nil {
+				return err
+			}
+			nestedPipeline.Processors = nestedProcessors
+		}
+		ddProcessor.Definition = nestedPipeline
+	case datadog.UrlParserType:
+		var urlParser = datadog.UrlParser{}
+		if sources, ok := tfProcessor["sources"].([]interface{}); ok {
+			urlParser.Sources = make([]string, len(sources))
+			for i, source := range sources {
+				urlParser.Sources[i] = source.(string)
+			}
+		}
+		if target, ok := tfProcessor["target"].(string); ok {
+			urlParser.Target = &target
+		}
+		if normalizedEndingSlashes, ok := tfProcessor["normalize_ending_slashes"].(bool); ok {
+			urlParser.NormalizeEndingSlashes = &normalizedEndingSlashes
+		}
+		ddProcessor.Definition = urlParser
+	case datadog.UserAgentParserType:
+		var userAgentParser = datadog.UserAgentParser{}
+		if sources, ok := tfProcessor["sources"].([]interface{}); ok {
+			userAgentParser.Sources = make([]string, len(sources))
+			for i, source := range sources {
+				userAgentParser.Sources[i] = source.(string)
+			}
+		}
+		if target, ok := tfProcessor["target"].(string); ok {
+			userAgentParser.Target = &target
+		}
+		if isEncoded, ok := tfProcessor["is_encoded"].(bool); ok {
+			userAgentParser.IsEncoded = &isEncoded
+		}
+		ddProcessor.Definition = userAgentParser
+	default:
+		return fmt.Errorf("failed to recoginize processor type: %s", processorType)
+	}
+
+	return nil
+}
+
+func buildFilter(d *schema.ResourceData, ddFilter *datadog.FilterConfiguration) {
+	if v, ok := d.Get("filter").([]interface{}); ok && len(v) > 0 {
+		convertFilter(v[0], ddFilter)
+	}
+}
+
+func convertFilter(tfFilter interface{}, ddFilter *datadog.FilterConfiguration) {
```

Comment:
same remarks as above, we could return to the result and prefer `to` to `convert` as it's shorter and used broadly in most of the programming languages